packages/core/src/components/forms/numericInput.tsx

```diff
@@ -421,19 +421,15 @@ export class NumericInput extends AbstractPureComponent2<HTMLInputProps & Numeri
 
     private renderInput() {
         const inputGroupHtmlProps = removeNonHTMLProps(this.props, NON_HTML_PROPS, true);
-        const valueAsNumber = this.getCurrentValueAsNumber();
-        const hasSpinButtons = this.props.buttonPosition !== undefined && this.props.buttonPosition !== "none";
 
         return (
             <InputGroup
                 asyncControl={this.props.asyncControl}
                 autoComplete="off"
                 id={this.numericInputId}
-                role={hasSpinButtons ? "spinbutton" : "textbox"}
+                role={this.props.allowNumericCharactersOnly ? "spinbutton" : "textbox"}
+                type={this.props.allowNumericCharactersOnly ? "number" : "text"}
                 {...inputGroupHtmlProps}
-                aria-valuemax={this.props.max}
-                aria-valuemin={this.props.min}
-                aria-valuenow={valueAsNumber}
```

Comment:
Realized that these are probably not needed due to the fact that the `<input>` already has `min` and `max` props, and a value.

Reply:
I don't think it hurts, just unneccessary

Reply:
Does the browser read min/max attributes when the input type is "text"? I feel like they could potentially be useful there... but again, I don't have specific documentation to support this idea.

Reply:
that is a very good point actually. We should probably keep these since the type is "text".